File: Best_Time_to_Buy_and_Sell_Stock.cpp
//https://leetcode.com/problems/best-time-to-buy-and-sell-stock/description/

class Solution {
public:
    int maxProfit(vector<int>& v) {
        int minm=v[0];
        int profit=0;
        for(int i=0;i<v.size();i++){
            minm= min(minm,v[i]);
            int current=v[i]-minm;
            profit=max(profit,current);
        }
        return profit;
    }
};

File: Best_Time_to_Buy_and_Sell_Stock.py
#https://leetcode.com/problems/best-time-to-buy-and-sell-stock/description/

class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        l,r=0,1
        maxp=0
        n=len(prices)
        while(r<n):
            if(prices[l]<prices[r]):
                profit=prices[r]-prices[l]
                maxp=max(profit,maxp)
            else:
                l=r
            r+=1
        return maxp
